docs/user-guide/data_types.rst

```diff
@@ -1,34 +1,54 @@
-Data types
-==========
+Array data types
```

Comment:
These docs are great. Since the TOC order is defined elsewhere and not modified in this PR, I wanted to suggest we relocate this page to the `Advanced Topics` section in the User Guide. Most users will not need to know 98% of this in order to use Zarr (which is good).